pkg/controller/topology_test.go

```diff
@@ -1323,6 +1323,44 @@ func TestPreferredTopologies(t *testing.T) {
 			},
 			expectError: true,
 		},
+		"allowedTopologies is subset of selected node's topology": {
+			allowedTopologies: []v1.TopologySelectorTerm{
+				{
+					MatchLabelExpressions: []v1.TopologySelectorLabelRequirement{
+						{
+							Key:    "com.example.csi/zone",
+							Values: []string{"zone1"},
+						},
+					},
+				},
+			},
+			nodeLabels: []map[string]string{
+				{"com.example.csi/zone": "zone1", "com.example.csi/rack": "rack1"},
+				{"com.example.csi/zone": "zone2", "com.example.csi/rack": "rack2"},
+				{"com.example.csi/zone": "zone2", "com.example.csi/rack": "rack1"},
+			},
+			topologyKeys: []map[string][]string{
+				{testDriverName: []string{"com.example.csi/zone", "com.example.csi/rack"}},
+				{testDriverName: []string{"com.example.csi/zone", "com.example.csi/rack"}},
+				{testDriverName: []string{"com.example.csi/zone", "com.example.csi/rack"}},
+			},
+			expectError: false,
+			expectedPreferred: []*csi.Topology{
+				{
+					Segments: map[string]string{
+						"com.example.csi/zone": "zone1",
```

Comment:
For the case where selectedNode is set, I would expect the fully qualified topology to be generated (with zone + rack) in strict and non-strict cases.

Reply:
That was case introduced with strict-topology only (since 1.4). Which we can use to put strictness on the topology information to pass  fully qualified topology list.

If we want to change that dehavior and nothing will break because of this, then I can make that change but would prefer a separate PR for that as this PR focuses on "subset of the leys are not working". WDYT?

Reply:
Ok, let's fix it in a followup PR.